### ISSUE TITLE
Video.js produces invalid IDs

### ISSUE THREAD

**Issue by mryand** [videojs/video.js#5167]
## Description
Video.js produces invalid IDs. Specifically, IDs that contain spaces.

## Steps to reproduce
Explain in detail the exact steps necessary to reproduce the issue.
1. use video.js
2. Inspect the dom
3. Note that there are invalid IDs present.

### Expected
I expect to see all the IDs on elements to be valid. Specifically, they should not contain spaces. Where IDs are being applied algorithmically, they should be passed through a sanitization routine that ensure that invalid characters are removed or replaced with valid characters.

### Actual
I see invalid IDs. Specifically, the IDs have spaces in them, like `"captions-font-family-vjs_video_3_component_379-Proportional Sans-Serif"`.

```html
<select aria-labelledby="captions-font-family-vjs_video_3_component_379">
    <option id="captions-font-family-vjs_video_3_component_379-Proportional Sans-Serif" value="proportionalSansSerif" aria-labelledby="captions-font-family-vjs_video_3_component_379 captions-font-family-vjs_video_3_component_379-Proportional Sans-Serif" style>
        Proportional Sans-Serif
    </option>
    <option id="captions-font-family-vjs_video_3_component_379-Monospace Sans-Serif" value="monospaceSansSerif" aria-labelledby="captions-font-family-vjs_video_3_component_379 captions-font-family-vjs_video_3_component_379-Monospace Sans-Serif" style>Monospace Sans-Serif</option>
    <option id="captions-font-family-vjs_video_3_component_379-Proportional Serif" value="proportionalSerif" aria-labelledby="captions-font-family-vjs_video_3_component_379 captions-font-family-vjs_video_3_component_379-Proportional Serif" style>Proportional Serif</option>
    <option id="captions-font-family-vjs_video_3_component_379-Monospace Serif" value="monospaceSerif" aria-labelledby="captions-font-family-vjs_video_3_component_379 captions-font-family-vjs_video_3_component_379-Monospace Serif" style>Monospace Serif</option>
    <option id="captions-font-family-vjs_video_3_component_379-Casual" value="casual" aria-labelledby="captions-font-family-vjs_video_3_component_379 captions-font-family-vjs_video_3_component_379-Casual" style>Casual</option>
    <option id="captions-font-family-vjs_video_3_component_379-Script" value="script" aria-labelledby="captions-font-family-vjs_video_3_component_379 captions-font-family-vjs_video_3_component_379-Script" style>Script</option>
    <option id="captions-font-family-vjs_video_3_component_379-Small Caps" value="small-caps" aria-labelledby="captions-font-family-vjs_video_3_component_379 captions-font-family-vjs_video_3_component_379-Small Caps" style>Small Caps</option>
</select>
```
A full list of invalid IDs from a given video.js instance are:

* `captions-font-family-vjs_video_3_component_379-Proportional Sans-Serif`
* `captions-font-family-vjs_video_3_component_379-Monospace Sans-Serif`
* `captions-font-family-vjs_video_3_component_379-Proportional Serif`
* `captions-font-family-vjs_video_3_component_379-Monospace Serif`
* `captions-font-family-vjs_video_3_component_379-Small Caps`

### Error output
Video.js does not produce errors directly. But Siteimprove, a popular QA and A11y testing tool, reports ""Aria-labelledby" has incorrect (ID) reference" related to this issue. This highlights the fact that this is not merely a validation issue but has real-world a11y impacts. Namely, aria-labelledby uses spaces to separate multiple IDs. So spaces in an ID prevent this from working correctly.

**Comment by OwenEdwards** [comment]
Closing as a duplicate of #4688.

**Comment by jcbreel** [comment]
This seems to be an issue still.

[https://monosnap.com/file/7ktqePRZb04kkqLVyukAiNhWeyOKYw](https://monosnap.com/file/7ktqePRZb04kkqLVyukAiNhWeyOKYw)

**Comment by OwenEdwards** [comment]
@jcbreel which version of video.js are you using? This wasn't fixed until [v6.10.0](https://github.com/videojs/video.js/tree/v6.10.0) and [v7.0.0](https://github.com/videojs/video.js/tree/v7.0.0).

**Comment by jcbreel** [comment]
Latest from NPM which is 6.9.0. I saw the issue was fixed on v7.0.0 but I'm getting some errors while uglifying in webpack.

**Comment by gkatsev** [comment]
going to make a patch and make 6.10 latest today. Can you open a new issue for the uglify stuff?

**Comment by jcbreel** [comment]
Sure, thank you!
